### PR TITLE
ATI Analytics for CPS Pages

### LINKED ISSUE #3454: Add ATI component to the MAP page
**Is your feature request related to a problem? Please describe.**

As a WS product manager I'd like the ability to track media asset page metrics with ATI Analytics on page load.

**Describe the solution you'd like**

Metrics will be captured on WS_NEWS_LANGUAGES_TEST 598343 until we release to production at which point we will capture analytics through WS_NEWS_LANGUAGES 598342

Example Media Asset Page: https://www.test.bbc.com/pidgin/media-49529724

AReS Payload: https://ares-broker.api.bbci.co.uk/api/asset/pidgin/media-49529724

| ID | ATI Site ID | Value | Source | mandatory |
|-|-|-|-|-|
| Destination (L1) | 598342 | WS_NEWS_LANGUAGES (*on live)*|Simorgh| Yes 
| | 598343|WS_NEWS_LANGUAGES_TEST (*on test)*|Simorgh| Yes |
|Producer  Level 2 |70|Pidgin |Simorgh| Yes |
|Chapter1|- |embedded_media (but can be world, news, embedded media - will generally be the second part of the page countername e.g. pidgin.embedded_media.media_asset.49529724.page ) |AReS| Yes |
|Page| -|pidgin.embedded_media.media_asset.49529724.page ("counterName" in payload) |AReS | Yes |
|*Custom Site Variables*| | | |
|Content ID (x1) |-|[urn:bbc:cps:c6f77da9-be7c-fb40-9a26-08fadbaede29] (this is in AReS payload in see "curie":"http://www.bbc.co.uk/asset/*c6f77da9-be7c-fb40-9a26-08fadbaede29*"|AReS| Yes |
|App Type (x2) |-|responsive OR amp |Simorgh| Yes |
|App name |-|news-pidgin|Simorgh | Yes | 
|Language|-|pcm |AReS| Yes | 
|URL|-|https://www.test.bbc.com/pidgin/media-49529724|Simorgh| Yes | 
|Content Type |-|article-media-asset|Simorgh| Yes | 
|Page Title (x9) |-|[BBC Pidgin Minute - BBC News Pidgin] (Defined as \{title} - BBC News Pidgin where content title is title from AReS payload)  | AReS for title / Simorgh for Brand | Yes | 
|Publication Date (x11) | -|[2018-05-23T05:54:44+00:00](firstPublished but needs to be converted from unix time)| AReS| Yes | 
|Update Date (x12) | -|[2018-05-23T05:54:44+00:00](lastUpdated but needs to be converted from unix time)|AReS | Yes | 
|Campaign (x16) | -|[WS - Inspire me] (Can be multiple values see: Campaigns / campaignName in payload) |AReS| No (can be blank) | 
| Category (x17) | -|[Feature] (defined as categoryName in the payload e.g. [news] [feature]) | AReS | No (can be blank) | 

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

**Test against these URLs:** 
      '/japanese/video-23248670',
         '/korean/media-23248686',
         '/pidgin/23248703',
         '/punjabi/media-23248705',
         '/sinhala/23248970',
         '/uzbek/sport-23248721',
         '/thai/thailand-23248713',
         '/news/av/business-23249447'

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**
Jira story - WSMEDIA-59

### PR DESCRIPTION
**Pull request by ryanmccombe**
Resolves https://github.com/bbc/simorgh/issues/3454

**Overall change:** Add ATI analytics tracking to the CPS page

**Code changes:**

- Add ATI analytics container and set up its data dependencies
- Add unit tests
- Bump bundle size (hit Persian threshold - 586KB)

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: scripts/bundleSizeConfig.js
Comment (ryanmccombe) [inline scripts/bundleSizeConfig.js:5]: Doesn't increase the bundles this much - nobody keeps these up to date.  This PR tipped over 586k, so reset the upper limit to +5kb

Comment (pjlee11) [reply scripts/bundleSizeConfig.js:5]: does the lower limit need increasing as well?

File: src/app/containers/ATIAnalytics/atiUrl/index.js
Comment (pjlee11) [inline src/app/containers/ATIAnalytics/atiUrl/index.js]: Could be more readable if `/ /g` was:

`const spaceRegex = / /g` then `.replace(spaceRegex, '%20')`

File: scripts/bundleSizeConfig.js
Comment (ryanmccombe) [reply scripts/bundleSizeConfig.js:5]: Yep - smallest bundle is sport (we're doing sport?!) at 555k

File: scripts/bundleSizeConfig.js
Comment (ryanmccombe) [inline scripts/bundleSizeConfig.js]:
```suggestion
   MIN_SIZE: 550,
```

File: src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js
Comment (pjlee11) [inline src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js:5]: Is there a reason we are passing `requestContext` and `serviceContext` as props rather than `useContext` here?

File: src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js
Comment (pjlee11) [inline src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js]: Is this comment necessary? Could the code not be more self documenting with something like:

```
const getChapter1 = page => page.split('.')[1];
...

const chapter1 = isValidPage ? getChapter1(page) : false; (I think it's false if not valid)
```

File: src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js
Comment (pjlee11) [inline src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js]: Will this error if `metadata` is undefined from the destructing on line 13? If so it can be `path()` ... same for `metadata.language`

File: src/app/containers/CpsAssetPageMain/__snapshots__/index.test.jsx.snap
Comment (pjlee11) [inline src/app/containers/CpsAssetPageMain/__snapshots__/index.test.jsx.snap:298]: Is this noscript expected?

Comment (ryanmccombe) [reply src/app/containers/CpsAssetPageMain/__snapshots__/index.test.jsx.snap:298]: Yep, it's part of the analytics.  It's there on the articles page too after they added the ATIAnalytics component

File: src/app/containers/ATIAnalytics/atiUrl/index.js
Comment (pjlee11) [inline src/app/containers/ATIAnalytics/atiUrl/index.js:155]: Do we have test cases that capture `campaign` being an array with the values joined by `~`?

Comment (ryanmccombe) [reply src/app/containers/ATIAnalytics/atiUrl/index.js:155]: Yep - https://github.com/bbc/simorgh/pull/4222/files#diff-92bd8e49974d9252cb4eff53ee71b19aR53

**Comment by ryanmccombe** [comment]
I can't merge this as it has unsigned commits - I also can't update it as master has unsigned commits :(

Already getting CRs here, so I'll leave it open - I'll squash it into a new branch once it is approved

File: src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js
Comment (ryanmccombe) [reply src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js:5]: Just the way the architecture is set up:  https://github.com/bbc/simorgh/blob/latest/src/app/containers/ATIAnalytics/params/index.js#L35-L41

All the page types are doing this

**Review by jroebu14 — APPROVED** [review]

File: src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js
Comment (chris-hinds) [reply src/app/containers/ATIAnalytics/params/cpsAssetPage/buildParams.js:5]: hmmm this does seem a little odd as to why we wouldn't be using useContext(), this would stop this PR but is something we should look at

**Review by chris-hinds — APPROVED** [review]

**Comment by ryanmccombe** [comment]
Don't merge this as it has unsigned commits

Branch here: https://github.com/bbc/simorgh/pull/4281

**Pull request closed without merging**
